sdk/servicebus/service-bus/src/shared/openLink.ts

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import * as log from "../log";
+import { Receiver as RheaReceiver, AwaitableSender } from "rhea-promise";
+import { translate, defaultLock } from "@azure/core-amqp";
+import { AbortSignalLike } from "@azure/core-http";
+import { AbortError } from "@azure/abort-controller";
+
+export type RheaLink = AwaitableSender | RheaReceiver;
+
+/**
+ * This is transitional code - as more code gets moved out of MessageReceiver/MessageSender
+ * we can stop passing in some of these functions.
+ *
+ * For instance
+ * - negotiateClaim + ensureTokenRenewal can be lifted out of LinkEntity
+ * - openLock can be moved into here if we allow openLink to have some manageable state
+ *  - same for isConnecting, which might need to be readable external from here but not writable.
+ *
+ * @internal
+ * @ignore
+ */
+export interface OpenArgs<RheaLinkT extends RheaLink> {
+  create(): Promise<RheaLinkT>;
+  isOpen(): boolean;
+
+  // logging related stuff.
+  logger: typeof log.sender;
+  logPrefix: string;
+
+  // TODO: I'd like to let openLink have some state (object?) and then these would be internal details
+  // we could still let external callers check if isConnecting is set, but not change it.
+  openLock: string;
+
+  isConnecting(): boolean;
+  setIsConnecting(value: boolean): boolean;
+
+  getCloseInitiated(): boolean;
+
+  // these are more candidates to move out from LinkEntity. They don't really depend on anything special
+  // in LinkEntity and are just clutter.
+  negotiateClaim(): Promise<void>;
+  ensureTokenRenewal(): Promise<void>;
```

Comment:
After looking into `_ensureTokenRenewal` in `LinkEntity`, its not clear to me why this would be an async function. All it does is set a timer to negotiate claim